pkg/skaffold/kubectl/exec_windows.go

```diff
@@ -42,54 +45,58 @@ func CommandContext(ctx context.Context, name string, arg ...string) *Cmd {
 }
 
 // Start starts the specified command in a job object but does not wait for it to complete
-func (c *Cmd) Start() (err error) {
-	var handle windows.Handle
-	handle, err = windows.CreateJobObject(nil, nil)
+func (c *Cmd) Start() error {
+	handle, err := windows.CreateJobObject(nil, nil)
 	if err != nil {
-		return
+		return fmt.Errorf("unable to create job: %w", err)
 	}
 
-	go func() {
-		<-c.ctx.Done()
-		c.Terminate()
-	}()
-
 	// https://gist.github.com/hallazzang/76f3970bfc949831808bbebc8ca15209
 	info := windows.JOBOBJECT_EXTENDED_LIMIT_INFORMATION{
 		BasicLimitInformation: windows.JOBOBJECT_BASIC_LIMIT_INFORMATION{
 			LimitFlags: windows.JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE,
 		},
 	}
-	if _, err = windows.SetInformationJobObject(
+	if _, err := windows.SetInformationJobObject(
 		handle,
 		windows.JobObjectExtendedLimitInformation,
 		uintptr(unsafe.Pointer(&info)),
 		uint32(unsafe.Sizeof(info))); err != nil {
-		return
+		return fmt.Errorf("unable to set job object: %w", err)
 	}
 
-	if err = c.Cmd.Start(); err != nil {
-		return
+	c.handleLock.Lock()
+	defer c.handleLock.Unlock()
+
+	if err := c.Cmd.Start(); err != nil {
+		return fmt.Errorf("unable to start: %w", err)
 	}
 
-	if err = windows.AssignProcessToJobObject(
+	if err := windows.AssignProcessToJobObject(
 		handle,
-		windows.Handle((*process)(unsafe.Pointer(c.Process)).Handle)); err != nil {
-		return
+		windows.Handle((*process)(unsafe.Pointer(c.Cmd.Process)).Handle)); err != nil {
+		return fmt.Errorf("unable assign process to job object: %w", err)
 	}
+
 	c.handle = handle
-	return
-}
 
-// Run starts the specified command in a job object and waits for it to complete
-func (c *Cmd) Run() error {
-	if err := c.Start(); err != nil {
-		return err
-	}
-	return c.Wait()
+	go func() {
+		<-c.ctx.Done()
+		c.Terminate()
+	}()
+
+	return nil
 }
 
 // Terminate closes the job object handle which kills all connected processes
 func (c *Cmd) Terminate() error {
-	return windows.CloseHandle(c.handle)
+	c.handleLock.Lock()
+	handle := c.handle
```

Comment:
It looks like it's possible for `handle` to be unset.  I suppose `windows.CloseHandle()` will error in that case?